hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/web/ozShell/OzoneAddress.java

```diff
@@ -124,6 +126,41 @@ public OzoneClient createClient(OzoneConfiguration conf)
     return client;
   }
 
+  /**
+   * Create OzoneClient for S3Commands.
+   * @param conf
+   * @param omServiceID
+   * @return OzoneClient
+   * @throws IOException
+   * @throws OzoneClientException
+   */
+  public OzoneClient  createClientForS3Commands(OzoneConfiguration conf,
+      String omServiceID)
+      throws IOException, OzoneClientException {
+    if (omServiceID != null) {
+      // OM HA cluster
+      if (OmUtils.isOmHAServiceId(conf, omServiceID)) {
+        return OzoneClientFactory.getRpcClient(omServiceID, conf);
+      } else {
+        throw new OzoneClientException("Service ID specified does not match" +
+            " with " + OZONE_OM_SERVICE_IDS_KEY + " defined in the " +
+            "configuration. Configured " + OZONE_OM_SERVICE_IDS_KEY + " are" +
+            conf.getTrimmedStringCollection(OZONE_OM_SERVICE_IDS_KEY));
+      }
+    } else {
+      // If om service id is not specified, consider it as a non-HA cluster.
```

Comment:
Can we handle the HA case where om service ID is not provided by CLI argument but local conf has a valid service ID without the Exception?

Reply:
I too thought that, but later I got a doubt that if conf has service ID, but it is of the remote cluster. (I think this can happen when conf has remote HA cluster details and the local non-HA cluster) . So, thrown exception and have marked this mandatory for HA cluster. Let me know your view on this.

Reply:
And also it can have multiple serviceID's when it has conf of local HA cluster and remote HA cluster.

Reply:
I think this approach makes sense.

Reply:
agree. Thanks for the context.